feconf.py

```diff
@@ -43,23 +43,28 @@
 TESTS_DATA_DIR = os.path.join('core', 'tests', 'data')
 SAMPLE_EXPLORATIONS_DIR = os.path.join('data', 'explorations')
 SAMPLE_COLLECTIONS_DIR = os.path.join('data', 'collections')
-INTERACTIONS_DIR = os.path.join('extensions', 'interactions')
-GADGETS_DIR = os.path.join('extensions', 'gadgets')
-RTE_EXTENSIONS_DIR = os.path.join('extensions', 'rich_text_components')
+INTERACTIONS_DIR = os.path.join(
+    'build', 'default', 'extensions', 'interactions')
+GADGETS_DIR = os.path.join('build', 'default', 'extensions', 'gadgets')
+RTE_EXTENSIONS_DIR = os.path.join(
+    'build', 'default', 'extensions', 'rich_text_components')
 
-OBJECT_TEMPLATES_DIR = os.path.join('extensions', 'objects', 'templates')
+OBJECT_TEMPLATES_DIR = os.path.join(
+    'build', 'default', 'extensions', 'objects', 'templates')
 
 # Choose production template if minification flag is used or
 # if in production mode
 TEMPLATES_DIR_PREFIX = 'prod' if (IS_MINIFIED or not DEV_MODE) else 'dev'
 FRONTEND_TEMPLATES_DIR = os.path.join(
     'core', 'templates', TEMPLATES_DIR_PREFIX, 'head')
-DEPENDENCIES_TEMPLATES_DIR = os.path.join('extensions', 'dependencies')
-VALUE_GENERATORS_DIR = os.path.join('extensions', 'value_generators')
+DEPENDENCIES_TEMPLATES_DIR = os.path.join(
+    'build', 'default', 'extensions', 'dependencies')
+VALUE_GENERATORS_DIR = os.path.join(
+    'build', 'default', 'extensions', 'value_generators')
 OBJECT_DEFAULT_VALUES_FILE_PATH = os.path.join(
-    'extensions', 'interactions', 'object_defaults.json')
+    'build', 'default', 'extensions', 'interactions', 'object_defaults.json')
 RULES_DESCRIPTIONS_FILE_PATH = os.path.join(
-    os.getcwd(), 'extensions', 'interactions', 'rules.json')
+    os.getcwd(), 'build', 'default', 'extensions', 'interactions', 'rules.json')
```

Comment:
Actually, I mostly disagree with this approach. It would be better to have the backend not need to read and assemble the interaction html/js -- maybe serve all of them and cache the resources heavily, or something. The backend is reading some (all?) of these files because it needs to compile things like  "interaction_templates", 'gadget_templates', etc., but the measure of success with externalizing things is removing the "interaction_templates", 'gadget_templates', etc. jinja variables altogether, caching pretty much all the data that they're currently sending, and reworking the codebase so as _not_ to manually assemble everything each time. If you do this right, I think we don't need to have files that are both static and application-readable. Let me know if I've interpreted this wrongly.

Also, it might be worth writing a brief design doc for this in order to get comments/agreement on the approach, so as not to have unnecessary back-and-forth.

Also, /cc @BenHenning.